spec/genre_spec.rb

```diff
@@ -0,0 +1,63 @@
+require_relative '../classes/genre'
+
+describe Genre do
+  let(:genre_name) { 'Action' }
+  let(:genre) { Genre.new(genre_name) }
+
+  describe '#initialize' do
+    it 'creates a new genre object with a random id and name' do
+      expect(genre.id).to be_a(Integer)
+      expect(genre.name).to eq(genre_name)
+      expect(genre.items).to be_empty
+    end
+  end
+
+  describe '.list_all_genre' do
+    it 'returns an array of all genre objects' do
+      expect(Genre.list_all_genre).to be_an(Array)
+    end
+  end
+
+  describe '.create_genre' do
+    it 'creates a new genre object if it does not already exist' do
+      new_genre_name = 'Horror'
+      new_genre = Genre.create_genre(new_genre_name)
+      expect(new_genre).to be_a(Genre)
+      expect(new_genre.name).to eq(new_genre_name)
+      expect(Genre.list_all_genre).to include(new_genre)
+    end
+
+    it 'returns an existing genre object if it already exists' do
+      # existing_genre = Genre.create_genre(genre_name)
```

Comment:
- [ ] Same here, Please do not leave unused code as a comment in your codebase. this will make your code much noisier, It is always a good practice to keep your code clean from such comments. 😉 👍🏻